app/liquid_tags/wikipedia_tag.rb

```diff
@@ -0,0 +1,99 @@
+class WikipediaTag < LiquidTagBase
+  PARTIAL = "liquids/wikipedia".freeze
+
+  def initialize(tag_name, input, tokens)
+    super
+    @data = get_data(input.strip)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: {
+        title: @data[:title],
+        extract: @data[:extract],
+        url: @data[:url]
+      },
+    )
+  end
+
+  private
+
+  def valid_url?(input)
+    input.match?(%r{^https?://([a-z-]+)\.wikipedia.org/wiki/([\S]+)$})
+  end
+
+  def get_data(input)
+    url = ActionController::Base.helpers.strip_tags(input).strip
+    raise StandardError, "Invalid Wikipedia share URL" unless valid_url?(url)
+
+    uri = URI.parse(url)
+    lang = uri.host.split(".", 2).first
+    title = uri.path.split("/").last
+    anchor = uri.fragment
+
+    if anchor
+      parse_page_with_anchor(url, lang, title, anchor)
+    else
+      parse_page(url, lang, title)
+    end
+  end
+
+  def parse_page_with_anchor(url, lang, title, anchor)
+    response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/mobile-sections/#{title}", format: :json)
+    handle_response_error(response, url)
+
+    text, section_title = get_section_contents(response, anchor, url)
+    title = "#{response['lead']['normalizedtitle']} - #{section_title}"
+
+    {
+      title: title,
+      url: url,
+      extract: text_clean_up(text)
+    }
+  end
+
+  def parse_page(url, lang, title)
+    response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/summary/#{title}", format: :json)
+    handle_response_error(response, url)
+
+    {
+      title: response["title"],
+      url: url,
+      extract: response["extract"]
+    }
+  end
+
+  def handle_response_error(response, input)
+    return if response.code == 200
+
+    raise StandardError, "Couldn't find the Wikipedia article {% #{tag_name} #{input} %}: #{response['detail']}"
```

Comment:
I don't think we need to render the entire tag here, you can just render the wrong URL and the error detail